### PR TITLE
Import of xml configuration for zonesets

### PR DESCRIPTION
**Pull request by agutenkunst**
Allows importing the generated xml configuration to supply the shape of the zonesets to the user.

## Open ToDos
- [x] Implement Parser + Tests
- [x] Make Parser robust towards edge cases / unexpected cases in the xml file
- [x] Wrap ROS around the parser and create a node publishing the zone information on a latched topic
- [x] Extend test for message content of integrationtest_config_server_node.cpp WIP @martiniil
- [x] Refactor parser into smaller functions
- [x] Create builder for ROS message @martiniil 
- [x] Solve conflicts @martiniil 

## Try Out
```
rosrun psen_scan_v2 config_server_node _config_file:=<path_to_config> _frame_id:="testframe"
```
---

<details>
<summary>PR Checklist</summary>

### Things to add, update or check by the maintainers before this PR can be merged.

- [x] Public api function documentation
- [x] ~Architecture documentation reflects actual code structure~
- [x] ~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~
- [x] Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)
- [x] Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))
- [x] Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))
- [x] CHANGELOG.rst updated
- [x] Copyright headers
- [x] Examples

### Review Checklist
- [x] Soft- and hardware architecture (diagrams and description)
- [x] Test review (test plan and individual test cases)
- [x] Documentation describes purpose of file(s) and responsibilities
- [x] Code (coding rules, style guide)

### Release planning (please answer)
- [x] When is the new feature released? **After visualization is done. Not yet.**
- [x] Which dependent packages do have to be released simultaneously? None

### Hardware tests
_Unstrike the text below to enable automatic hardware tests if available. Otherwise use this as a request for the reviewer._
- [x] ~Perform hardware tests~

</details>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -347,6 +393,18 @@ if(CATKIN_ENABLE_TESTING)
...
+    standalone/test/unit_tests/configuration/unittest_xml_configuration_parser.cpp
+  )
+  target_link_libraries(unittest_xml_configuration_parser
+    ${catkin_LIBRARIES}
+    ${PROJECT_NAME}_standalone_xml_configuration_import
+    ${console_bridge_LIBRARIES}
+    fmt::fmt
+  )
+  file(COPY standalone/test/unit_tests/configuration/unittest_xml_configuration_parser-testfile-no-speedrange.xml DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/testfiles)
+  file(COPY standalone/test/unit_tests/configuration/unittest_xml_configuration_parser-testfile-with-speedrange.xml DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/testfiles)
```
Comment (rfeistenauer) [inline CMakeLists.txt:406]:
```suggestion
  file(COPY standalone/test/unit_tests/configuration/unittest_xml_configuration_parser-testfile-no-speedrange.xml 
       DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/testfiles)
  file(COPY standalone/test/unit_tests/configuration/unittest_xml_configuration_parser-testfile-with-speedrange.xml 
       DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/testfiles)
```

File: include/psen_scan_v2/config_server_node.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#include "psen_scan_v2/ZoneSetConfiguration.h"
+
+using namespace std;
+
+/**
+ * @brief Root namespace for the ROS part
+ */
+namespace psen_scan_v2
+{
+static const std::string DEFAULT_ZONESET_TOPIC = "zonesets";
```
Comment (rfeistenauer) [inline include/psen_scan_v2/config_server_node.h:35]: Thought the topic name is something like "config" or "zoneset_config"?

File: msg/ZoneSet.msg [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Bezieht sich auf eine Zone, refenziert Scanner und frame
```
Comment (rfeistenauer) [inline msg/ZoneSet.msg:1]: german

File: scripts/zone_configuration_server.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python3
```
Comment (rfeistenauer) [inline scripts/zone_configuration_server.py:1]: We wount be needing this in melodic and earlier?

File: scripts/zone_configuration_server.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
```
Comment (rfeistenauer) [inline scripts/zone_configuration_server.py:2]: copyright

File: scripts/zone_configuration_server.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import rospy
+
+def server():
+    x=0xD80E
+    x=((x << 8) | (x >> 8)) & 0xFFFF
+    print(x)
```
Comment (rfeistenauer) [inline scripts/zone_configuration_server.py:8]: do i have to understand that?

File: scripts/zone_switching_visualization.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
```
Comment (rfeistenauer) [inline scripts/zone_switching_visualization.py:2]: copyright

File: standalone/include/psen_scan_v2_standalone/configuration/xml_configuation_parser.h [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+ * @brief Convert string from a \<ro\> element to values.
+ *
+ * The value in a \<ro\> element is a string with length 4*N where N is the number of distance values.
+ * 4 succedding values form a set that can be transformed into the lenth in mm.
+ * The conversion of 4 characters follows the rule "abcd" -> 0xcdab
+ * Examples:
+ * "D307" -> 0x07D3 -> 2003mm
+ * "ED03" -> 0x03ED -> 1005mm
+ * "2B01" -> 0x012B -> 299mm
+ * "8913" -> 0x1389 -> 5001mm
```
Comment (rfeistenauer) [inline standalone/include/psen_scan_v2_standalone/configuration/xml_configuation_parser.h:56]: Would move this explanation to ro_value_to_uint where you really do this magic and refer to it with @ref or @see

File: standalone/src/configuration/xml_configuration_parser.cpp [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+{
+  ZoneSetConfiguration zoneset_config;
+
+  tinyxml2::XMLConstHandle docHandle(&doc);
+  tinyxml2::XMLConstHandle xml_set_info_handle = docHandle.FirstChildElement("MIB")
+                                                     .FirstChildElement("scannerDescr")
+                                                     .FirstChildElement("zoneSetDefinition")
+                                                     .FirstChildElement("zoneSetInfo");
+
+  // Loop over all <zoneSetInfo>
```
Comment (rfeistenauer) [inline standalone/src/configuration/xml_configuration_parser.cpp:37]: loose comments are confusing ^^

File: scripts/zone_switching_visualization.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
```
Comment (rfeistenauer) [inline scripts/zone_switching_visualization.py:1]: visualization only necessary from noetic onwards?

File: include/psen_scan_v2/zone_configuration_conversion.h [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+    polygon.points.push_back(point);
+
+    phi += phi_inc;
+  }
+
+  return polygon;
+}
+
+psen_scan_v2::ZoneSet toMsg(const ZoneSetStandalone& zoneset)
```
Comment (rfeistenauer) [inline include/psen_scan_v2/zone_configuration_conversion.h:51]: maybe call it `toRosMsg`
